仕様書のマークダウンを追加しました

iorar opened the pull request:
暫定の仕様やコーディング規約、マクロリストを載せたspecification.mdを追加しました. 
wikiの代わりに育てていくことを想定しています.
わかりやすい書式の提案があればいただきたいです.

iorar committed:
仕様書のマークダウンを追加しました

暫定の仕様やコーディング規約、マクロリストを載せたマークダウンを追加しました. wikiの代わりに育てていくことを想定しています.
わかりやすい書式の知見が無いので提案が欲しいです.

idomshi reviewed (commented):
- 追記履歴は書かない方が良いと思いました（更新を忘れそうなので）
- TODOや提案はissueに載せると議論もできて便利かも
- ``ending_text``は``text``パラメーターを取るので修正していただけると嬉
- あとで画像のサイズ・音声のビットレートとサンプリングレートを決めたい（いまは未定で良いと思います）

……すぐに決まらない項目はissueで個別に話し合いましょう。

iorar commented:
了解です，もろもろ修正してもう一度プルリクを投げておきます
追記履歴は確かになあなあになりそうなので書かないことにします
ビットレート他TODOについてはissueにて立て直しておきます

iorar committed:
Update specification.md

・仕様書というのもおかしいので、見出しは規約などまとめとした
・TODOに書きたい内容はissueに移管
・追記履歴は廃止
・マクロ[ending_text]のパラメータ`text`を追加

iorar committed:
add macro reference, coding rules

・規約周りはcoding_rules.mdに、マクロリファレンスはmacro_reference.mdに分割
・規約周りはとりあえず前回の話し合いの内容をほぼコピペ

iorar commented:
仕様書と称するのはかなり御幣があるので，ファイル名はcoding_rules.mdとmacro_reference.mdとしました．

iorar committed:
開発手法について追加　他

・GitHub Flowに従う旨を明記
・他、表現の微修正

iorar committed:
不定部分を確定

・素材の命名規則を確定
・音楽素材の形式についての規約を最終的にoggに変換する形に変更
